src/FrameManager.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  ClientToHost,
+  validate as validateIncoming
+} from './messages/ClientToHost';
+import { HostToClient } from './messages/HostToClient';
+
+const IFRAME_STYLE =
+  'position: absolute; top: 0; left: 0; width: 100%; height: 100%;';
+
+/**
+ * A handler function for messages sent from a client app.
+ */
+type MessageHandler = (event: ClientToHost) => void;
+
+/**
+ * FrameManager is responsible for managing the state of the client iframe.
+ * It changes locations in the frame when requested, and handles proxying
+ * of postMessage events to and from the client, with verification of
+ * the expected client origin.
+ */
+class FrameManager {
+  private _iframe: HTMLIFrameElement;
+  private _frameLocation: string;
+  private _window: Window;
+  private _postMessageHandler: (event: MessageEvent) => void;
+
+  constructor(options: { onMessage: MessageHandler; mockWindow?: Window }) {
+    this._window = options.mockWindow ? options.mockWindow : window;
+    this._postMessageHandler = event => {
+      this._handlePostMessage(options.onMessage, event);
```

Comment:
I had to change this up a little bit from the parent, since it wasn't actually using the subscription manager to check the interested topics.